core/src/replay_stage.rs

```diff
@@ -55,6 +55,7 @@ use std::{
 pub const MAX_ENTRY_RECV_PER_ITER: usize = 512;
 pub const SUPERMINORITY_THRESHOLD: f64 = 1f64 / 3f64;
 pub const MAX_UNCONFIRMED_SLOTS: usize = 5;
+pub const UNLOCK_SWITCH_VOTE_SLOT: Slot = 17_000_000;
```

Comment:
@mvines for integration, should I do what we did for the nonce changes, i.e.

1) Wait for 1.2 to cut, add  `UNLOCK_SWITCH_VOTE_SLOT` only to 1.2
2) Don't check in `UNLOCK_SWITCH_VOTE_SLOT` to master

Reply:
This looks like a softer change, not a hard ABI breakage.  We can't handle nodes producing the old/new voting behaviour?   These slot-based gates are a serious PITA to roll out across

Reply:
@carllin what if the switch was a “AddSwitchProof” instruction.  I think current nodes will simply fail to run that command but the vote can be registered

Reply:
@aeyakovenko you mean validators will have to submit two different transactions, a regular vote transaction and an "AddSwitchProof" instruction? 

That makes detection in gossip harder because you may receive those out of order, and you won't be able to distinguish which votes are switching/non-switching until you see both.

Reply:
@mvines The problem is the new voting behavior potentially submits a different vote instruction when switching forks, which will be ignored by validators who haven't upgraded. 

We also don't want them submitting both types of instructions because it's important that validators only submit one or the other to uphold the safety guarantees when/f optimistic confirmation is detected

Reply:
@carllin one TX, the new one has to instructions, vote and addswitchproof

Reply:
or, do this

PrepareSwitchProof(hash, vote_slot) -> just sets a local variable with the hash and slot
Vote(vote_slot) -> consumes the above

which would allow us to roll the update.

Reply:
ix, 1 would fail, but ix two would succeed.  Ah, i guess that wont work.  The return status is going to be different

Reply:
@aeyakovenko I added one tx with two instructioons, it seems to be ok?

Reply:
@aeyakovenko hmmm I just added the two instructions per vote tx (see latest changes), is there some issue?

Reply:
the corresponding `cluster_info_vote_listener` changes: https://github.com/solana-labs/solana/pull/10179/commits/e532d2152c652917ea8b37b99a525ce231f6cab3